service/resource/legacy/daemonset.go

```diff
@@ -3,38 +3,44 @@ package legacy
 import (
 	"github.com/giantswarm/flanneltpr"
 	apismeta "k8s.io/apimachinery/pkg/apis/meta/v1"
-	apisextensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
-
 	"k8s.io/apimachinery/pkg/util/intstr"
 	api "k8s.io/client-go/pkg/api/v1"
+	apisextensions "k8s.io/client-go/pkg/apis/extensions/v1beta1"
+
+	"github.com/giantswarm/flannel-operator/service/key"
 )
 
-func newDaemonSet(spec flanneltpr.Spec, etcdCAFile, etcdCrtFile, etcdKeyFile string) *apisextensions.DaemonSet {
+func newDaemonSet(customObject flanneltpr.CustomObject, etcdCAFile, etcdCrtFile, etcdKeyFile string) *apisextensions.DaemonSet {
 	app := networkApp
 
-	labels := map[string]string{
-		"cluster":  clusterName(spec),
-		"customer": clusterCustomer(spec),
-		"app":      app,
-	}
-
-	containers := newDaemonSetContainers(spec, etcdCAFile, etcdCrtFile, etcdKeyFile)
-	volumes := newDaemonSetVolumes(spec)
+	containers := newDaemonSetContainers(customObject.Spec, etcdCAFile, etcdCrtFile, etcdKeyFile)
+	volumes := newDaemonSetVolumes(customObject.Spec)
 
-	return &apisextensions.DaemonSet{
+	daemonSet := &apisextensions.DaemonSet{
 		TypeMeta: apismeta.TypeMeta{
 			Kind:       "daemonset",
 			APIVersion: "extensions/v1beta",
 		},
 		ObjectMeta: apismeta.ObjectMeta{
-			Name:   app,
-			Labels: labels,
+			Name: app,
+			Annotations: map[string]string{
+				VersionBundleVersionAnnotation: key.VersionBundleVersion(customObject),
+			},
+			Labels: map[string]string{
+				"app":      app,
+				"cluster":  clusterName(customObject.Spec),
+				"customer": clusterCustomer(customObject.Spec),
+			},
```

Comment:
I extracted the labels variable so we don't have to repeat that and they are in sync. I even think it makes sense to extract `key.Labels` or `key.BaseLabels`.